CHANGES/1512.bugfix.rst

```diff
@@ -0,0 +1,7 @@
+Aligned the type of exception raised for improper IPv6 URL values,
+which should be `ValueError`.
+
+It fixes the issue where exception `IndexError` was raised in the
+following conditions: empty IPv6 URL, brackets in reverse order.
```

Comment:
```suggestion
These fixes the issue where exception :exc:`IndexError` was
leaking from the internal code because of not being handled and
transformed into a user-facing error. The problem was happening
under the following conditions: empty IPv6 URL, brackets in
reverse order.
```